graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/SchemaTransformer.java

```diff
@@ -45,6 +49,12 @@ public SchemaTransformer fetchEntities(DataFetcher entitiesDataFetcher) {
         return this;
     }
 
+    @NotNull
+    public SchemaTransformer directivePredicate(Predicate<GraphQLDirective> directivePredicate) {
```

Comment:
This change really deserves to be in a separate PR - but you can find more background [here](https://github.com/leangen/graphql-spqr/issues/290#issuecomment-562025005).

Reply:
Yes I agree that it would be helpful for this to be in a separate PR because it's a separate bit of functionality.

If I'm following along correctly here it seems that the desire is to exclude the private directives because they cause the schema to fail to be transformed successfully. The failure is due to the fact that the directives are not included in the schema. Why is it desirable to use directives in a schema but not include the definition of the directives?

Reply:
> Why is it desirable to use directives in a schema but not include the definition of the directives?

I'm afraid I can't answer that question. I believe it's an implementation detail of graphql-spqr. Ping @kaqqao?

Reply:
This is actually part of the federation spec for the `sdl` field: https://www.apollographql.com/docs/apollo-server/federation/federation-spec/#query_service

We have to do the same logic here in graphql-kotlin-federation: https://github.com/ExpediaGroup/graphql-kotlin/blob/8ff7de80e8681b75091f244564a991aef7eea4fd/graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/federation/FederatedSchemaGeneratorHooks.kt#L90-L97

I have created a PR in graphql-java to make this a little easier by adding a new option to the printer: https://github.com/graphql-java/graphql-java/pull/1756

Reply:
There's been a slight misunderstanding about Apollo Federation, there's more info at  https://github.com/apollographql/federation-jvm/pull/41#issuecomment-599715362